src/main/java/org/broadinstitute/hellbender/tools/GetSampleName.java

```diff
@@ -95,12 +93,10 @@ public void onTraversalStart() {
             throw new UserException.BadInput("The given bam input has no sample names.");
         }
 
-        try (final FileWriter fileWriter = new FileWriter(outputSampleNameFile, false)) {
+        try (final PrintWriter fileWriter = new PrintWriter(outputSampleNameFile.getOutputStream(), false)) {
             final String rawSample = sampleNames.get(0);
             final String outputSample = urlEncode ? IOUtils.urlEncode(rawSample) : rawSample;
             fileWriter.write(outputSample);
-        } catch (final IOException ioe) {
-            throw new UserException("Could not write file.", ioe);
```

Comment:
If we're no longer calling methods that throw `IOException`, then I think we need to call `checkError()` manually on the `PrintWriter` to check whether an error occurred....

Reply:
So after reviewing this, I think you're right in this case that we need to call checkError now, since we went from FileWriter (throws `IOException`) to PrintWriter(does not).

Reply:
Actually, even better I think is to switch to restore the catch block and switch to OutputStreamWriter, which propagates IOExceptions.